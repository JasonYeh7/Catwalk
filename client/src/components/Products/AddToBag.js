import React from 'react';
import $ from 'jquery';

 var AddToBag = () => {
  return (
    <div className="AddToBag">
      <button className="AddToBagBtn">ADD TO BAG</button>
    </div>
  )
}

export default AddToBag;


// var addedToBag = (e) => {
//   e.preventDefault();
//   if ($(e.currentTarget).hasClass('bagAdd')) {
//    $(e.currentTarget).removeClass('bagAdd');
//   } else {
//    $(e.currentTarget).addClass('bagAdd');
//   }
// }