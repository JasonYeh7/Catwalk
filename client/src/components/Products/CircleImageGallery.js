import React from 'react';

 var CircleImageGallery = () => {
  return (
    <div>

    </div>
  )
}

export default CircleImageGallery;