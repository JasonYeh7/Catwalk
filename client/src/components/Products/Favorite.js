import React from 'react';

 var Favorite = () => {
  return (
    <div className="favorite">

    </div>
  )
}

export default Favorite;