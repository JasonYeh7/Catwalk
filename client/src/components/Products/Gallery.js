import React from 'react';

 var Gallery = (props) => {
  return (
    <div className="thumbnail">
      <img src={props.image} width="25" height="50" onClick={props.click}/>
    </div>
  )
}

export default Gallery;