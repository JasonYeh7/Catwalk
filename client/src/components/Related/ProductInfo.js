import React from 'react';
import Star from '../Star/Star.js'
import styles from './relatedItems.module.css'

const ProductInfo = ({product}) => {
  return (
    <div className={styles.productInfo}>
      <h5 className={styles.productText}>{product.category}</h5>
      <h4 className={styles.productText}>{product.name}</h4>
      <h5 className={styles.productText}>{product.default_price}</h5>
      <div className={styles.productText}>
        <Star rating={4.0}/>
      </div>
    </div>
  )
};

export default ProductInfo