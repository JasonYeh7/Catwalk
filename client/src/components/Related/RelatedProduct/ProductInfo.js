import React from 'react';
import Star from '../../Star/Star.js'
import styles from './relatedItems.module.css'

const ProductInfo = (props) => {
  let index = props.index + props.movement;
  let id = props.relatedId;
  let data = props.data;

  let category = data.dummyRelatedProducts.[index].category;
  let name = data.dummyRelatedProducts[index].name;
  let price = data.dummyRelatedProducts[index].default_price;
  return (
    <div className={styles.productInfo}>
      <h5 className={styles.productText}>{category}</h5>
      <h4 className={styles.productText}>{name}</h4>
      <h5 className={styles.productText}>{price}</h5>
      <div className={styles.productText}>
        <Star rating={4.0}/>
      </div>
    </div>
  )
};

export default ProductInfo