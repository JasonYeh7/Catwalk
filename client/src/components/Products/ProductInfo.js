import React from 'react';
import ProductTitle from './ProductTitle.js';
import SelectStyle from './SelectStyle.js';
import CircleImageGallery from './CircleImageGallery.js';
import SelectSize from './SelectSize.js';
import AddToBag from './AddToBag.js';
import Favorite from './Favorite.js';
import SelectQuantity from './SelectQuantity.js'
import Star from '../Star/Star.js';

 var ProductInfo = (props) => {
  return (
    <div className="product-info-container">
      <div className="product-star">
        <Star rating={3.5}/>
      </div>
      <ProductTitle />
      <SelectStyle />
      <ol className="circleImgBox">
      {
        props.images.map(image => {
          var imgid = image.split('/')[10];
          return <CircleImageGallery image={image} key={imgid} show={props.show}/>
        })
      }
      </ol>
      <SelectSize />
      <SelectQuantity />
      <AddToBag />
      <Favorite />
    </div>
  )
}

export default ProductInfo;