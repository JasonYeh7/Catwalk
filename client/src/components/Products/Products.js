import React from 'react';
import ThumbnailGallery from './ThumbnailGallery.js';
import MainImageView from './MainImageView.js';
import ProductInfo from './ProductInfo.js';

class Products extends React.Component {
  constructor() {
    super();
    this.state = {
      images: ['https://images.pexels.com/photos/4354221/pexels-photo-4354221.jpeg', 'https://images.pexels.com/photos/3310694/pexels-photo-3310694.jpeg', 'https://images.pexels.com/photos/2587392/pexels-photo-2587392.jpeg', 'https://images.pexels.com/photos/4614845/pexels-photo-4614845.jpeg'],
      selected: null
    };
    this.show = this.show.bind(this);
    this.fowardButton = this.fowardButton.bind(this);
    this.backButton = this.backButton.bind(this);
  }

  show(e) {
    this.setState({
      selected: e.currentTarget.src
    })
  }

  fowardButton(e) {
    e.preventDefault();
    var images = this.state.images
    var current = this.state.selected
    if (current === images[images.length - 1]) {
      this.setState({
        selected: images[0]
      });
      return;
    }
    for (var i = 0; i < images.length; i++) {
      if (images[i] === current) {
        this.setState({
          selected: images[i+1]
        });
      }
    }
  }

  backButton(e) {
    e.preventDefault();
    var images = this.state.images
    var current = this.state.selected
    if (current === images[0]) {
      this.setState({
        selected: images[images.length - 1]
      });
      return;
    }
    for (var i = 0; i < images.length; i++) {
      if (images[i] === current) {
        this.setState({
          selected: images[i-1]
        });
      }
    }
  }

  render() {
    return(
      <div className="container" style={{border: "solid"}}>
        <div className="thumbnail-slider">
          <div className="viewport">
            {
              this.state.images.map(image => {
                var imgid = image.split('/')[4];
                return  <ThumbnailGallery image={image} key={imgid} click={this.show}/>
              })
            }
          </div>
        </div>
        <div className="main-image-container">
          <MainImageView forward={this.fowardButton} back={this.backButton} select={this.state.selected}/>
        </div>
        <div className="product-info-right">
          <ProductInfo />
        </div>
      </div>
    );
  }
};

export default Products;