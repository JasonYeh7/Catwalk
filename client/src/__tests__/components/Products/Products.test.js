import React from 'react';
import { shallow, render, mount } from '../../../enzyme';
import Products from '../../../components/Products/Products.js';
import ProductInfo from '../../../components/Products/ProductInfo.js';

describe('Products Component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <Products />
    )
  });

  it('Renders the Products Component', () => {
    expect(wrapper.exists()).toBe(true);
  });

});


{/* <Products />
  <ThumbnailGallery/>
  <MainImageView />
  <ProductInfo />
    <Star />
    <ProductTitle />
    <SelectStyle />
    <CircleImageGallery />
    <SelectSize />
    <AddToBag />
    <Favorite /> */}
