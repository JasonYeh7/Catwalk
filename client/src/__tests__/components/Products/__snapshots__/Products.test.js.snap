// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products Component Should render correctly with no props 1`] = `
<ContextProvider
  value={null}
>
  <Products
    dispatch={[Function]}
    products={
      Object {
        "fullSizeImage": Array [],
        "maxQty": 1,
        "meta": Object {},
        "product": Object {
          "category": "",
          "default_price": 0,
          "description": "",
          "name": "",
        },
        "products": Array [],
        "qtyNSize": Array [
          Object {
            "id": Object {
              "quantity": 0,
              "size": "",
            },
          },
        ],
        "saleOrDefaultPrice": Object {
          "name": "",
          "price": 0,
          "sale": 0,
        },
        "saleOrNot": Array [
          Object {
            "name": "",
            "price": 0,
            "sale": 0,
          },
        ],
        "selected": null,
        "styleImageArr": Array [
          Object {
            "fullSizeImage": Array [],
            "name": "",
            "thumbNailImages": Array [],
          },
        ],
        "styles": Array [],
        "thumbNailImages": Array [],
      }
    }
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />
</ContextProvider>
`;
